rbc/omniscidb.py

```diff
@@ -281,6 +281,40 @@ def __init__(self,
         # An user-defined device-LLVM IR mapping.
         self.user_defined_llvm_ir = {}
 
+    def __call__(self, *args, **kwargs):
+        msg = """Decorated functions called directly. Decorated function are
+        not meant to be called directly, but are registered and later
+        executed directly on the database.
+
+        Following is a proper usage of `RemoteOmnisci`.
+
+        >>> from rbc.omniscidb import RemoteOmnisci
+        >>> omni = RemoteOmnisci()
+        >>> @omni('double(double)')
+        ... def farenheight2celcius(f):
+        ...     return (f - 32) * 5 / 9
+        ...
+        >>> _, r = omni.sql_execute('select farenheight2celcius(40)')
+        >>> print(list(r))
+        [(4.444444444444445,)]
+
+        Alternatively, you can define a Python function and decorate it
+        in a second step. This allows you to be able to call the Python
+        function. In any case, the decorated function cannot be called.
+
+        >>> omni = RemoteOmnisci()
+        >>> def farenheight2celcius(f):
+        ...     return (f - 32) * 5 / 9
+        ...
+        >>> decorated = omni('double(double)')(farenheight2celcius)
+        >>> _, r=omni.sql_execute('select farenheight2celcius(40)')
+        >>> print(list(r))
+        [(4.444444444444445,)]
+        >>> print(farenheight2celcius(40))
+        4.444444444444445
+        """
+        raise UnsupportedError(msg)
+
```

Comment:
wow, this is a a very good docstring :).
However, I'm not sure that it's good to have such a big and long error message. Maybe it's better to put this text in some online docs and put a link to it in the error message?

Reply:
Yes I can remove the good practice part. The initial thought was to ship this and then wait for the new doc to then link to it. Or can I already update some existing doc and link to it?

Reply:
ah ok, this approach makes sense, indeed